draft-ietf-anima-brski-prm.md

```diff
@@ -1815,58 +1908,76 @@ The content of this field is not subject to standardization, but examples are pr
  ~                  ~                 ~                 ~            ~
 (7) Supply CA Certificates to Pledge
  ~                  ~                 ~                 ~            ~
  |                  |                 |                 |            |
  |<----opt. TLS---->|                 |                 |            |
- |<-----cACerts-----|                 |                 |            |
+ |<-----caCerts-----|                 |                 |            |
  |                  |                 |                 |            |
  ~                  ~                 ~                 ~            ~
 ~~~~
 {: #exchangesfig_uc2_7 title="Certificate provisioning exchange" artwork-align="center"}
 
+The Registrar-Agent SHALL provide the bag of CA certificates requested from and signed by the registrar to the pledge by HTTP(S) POST to the pledge endpoint at `/.well-known/brski/scac`.
+The request body MUST contain the caCerts artifact as defined in {{cacerts-artifact}}.
+In the request header, the Content-Type field MUST be set to `application/jose+json`.
 
-### Request Artifact:
-
-The Registrar-Agent SHALL provide the set of CA certificates requested from the registrar to the pledge by HTTP POST to the endpoint: "/.well-known/brski/scac".
-
-As the CA certificate provisioning is crucial from a security perspective, this provisioning SHOULD only be done, if the voucher-response has been successfully processed by pledge as reflected in the voucher status telemetry.
-
-The CA certificates message has the Content-Type `application/jose+json` and is signed using the credential of the registrar as shown in {{PCAC}}.
+Upon receiving valid caCerts, the pledge MUST first verify the signature of the registrar using the initial trust anchor from the pinned domain certificate.
+If the pledge is unable to process the caCerts, it SHOULD respond with an HTTP error code.
+The following client error status codes SHOULD be used:
 
-The CA certificates are provided as base64-encoded "x5bag".
-The pledge SHALL install the received CA certificates as trust anchor after successful verification of the registrar's signature.
-
-
-### Response (no artifact)
+* 400 Bad Request: if the pledge detects an error in the format of the request, e.g., missing field, wrong data types, etc. or if the request is not valid JWS-signed JSON even though the Content-Type request header field was set to `application/jose+json`
+* 401 Unauthorized: if the signature cannot be verified against the installed initial trust anchor (pinned domain certificate)
+* 415 Unsupported Media Type: if the Content-Type request header field indicates a type that is unknown or unsupported, e.g., a type other than `application/jose+json`
 
 The verification comprises the following steps the pledge MUST perform. Maintaining the order of versification steps as indicated allows to determine, which verification has already been passed:
 
+TODO(steps 3 and 4 overlap)
```

Comment:
Steps 1 and 2 are already converted to the previous document style with a ist of given client error status codes.

Steps 3 and 4 do not make sense, as the artifact is signed by the registrar and the registrar LDevID cert is directly one of the trust anchors and often identical to the pinned domain cert.
It fails anyway if the CA certs come from a different registrar, if it is compared against the registrar LDevID cert taken from the Voucher signature.

Reply:
I guess the question may have been answered with the comment in row 1809

Reply:
I still do not understand Step 3, where the signing cert is compared directly against the pinned domain cert (MAY even compare hash directly):
If there are multiple registrars and an issuing or root cert is the pinned domain cert, then this fails although it is valid.

Unclear what Step 4 is verifying: "domain certificate contained in the voucher" is the pinned domain certificate from Step 3, no?

Reply:
Solved in call.